Client/Frontend/Home/FxHomeJumpBackInView.swift

```diff
@@ -0,0 +1,251 @@
+
+        if deviceIsiPad {
+            if horizontalSizeClassIsCompact { return verticalVariables }
+            return horizontalVariables
+
+        } else {
+            if deviceIsInLandscapeMode { return horizontalVariables }
+            return verticalVariables
+        }
+    }
+
+    lazy var collectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = layoutVariables.scrollDirection
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.isScrollEnabled = false
+        collectionView.backgroundColor = UIColor.clear
+        collectionView.clipsToBounds = false
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.register(JumpBackInCell.self, forCellWithReuseIdentifier: JumpBackInCell.cellIdentifier)
+
+        return collectionView
+    }()
+
+    // MARK: - Inits
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupLayout()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Helpers
+    private func setupLayout() {
+        contentView.addSubview(collectionView)
+
+        NSLayoutConstraint.activate([
+            collectionView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 2),
+            collectionView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -2),
+            collectionView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor)
+        ])
+    }
+}
+
+extension FxHomeJumpBackInCollectionCell: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        viewModel.updateDataAnd(layoutVariables)
+        return viewModel.jumpableTabs.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: JumpBackInCell.cellIdentifier, for: indexPath) as! JumpBackInCell
+
+        if let item = viewModel.jumpableTabs[safe: indexPath.row] {
+            let itemURL = item.url?.absoluteString ?? ""
+            let site = Site(url: itemURL, title: item.displayTitle, bookmarked: true)
+
+            profile?.favicons.getFaviconImage(forSite: site).uponQueue(.main, block: { result in
+                guard let image = result.successValue else { return }
+                cell.heroImage.image = image
+                cell.setNeedsLayout()
+            })
+
+            cell.itemTitle.text = site.title
+            // TODO: Determine source string here, if from synced tabs. Simply update `cell.itemDetails.text`
```

Comment:
Is it safe to assume this TODO is okay to leave for this PR?

Reply:
Leaving this TODO becasue this is actually part of a separate feature on this feature. I wasn't supposed to implement it here, but I figured it was easier to do the stackview work now, rather than leave it for later. So adding the appropriate string can be done later.